### PR TITLE
Datadog Incident Management guide

### PR DESCRIPTION
**Pull request by bernardjkim**
This PR adds a documentation guide for the Datadog Incident Management plugin that will be added with https://github.com/gravitational/teleport/pull/46271.

### CODE REVIEW

**Review by r0mant — COMMENTED** [review]
@bernardjkim Couple of comments:

- Can the guide be geared towards hosted version of the plugin rather than self-hosted? Similar to [Opsgenie](https://goteleport.com/docs/admin-guides/access-controls/access-request-plugins/opsgenie/) or [ServiceNow](https://goteleport.com/docs/admin-guides/access-controls/access-request-plugins/servicenow/). That's one less configuration to support.
- You need to update [config.js](https://github.com/gravitational/teleport/blob/master/docs/config.json) to add the new entry to table of contents on the left nav bar.

File: docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx
Comment (hugoShaka) [inline docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:16]: It's not clear if this is a cloud-only feature, enteprise-only, or can still be done by running the self-hosted plugin (we do have [the main function here](https://github.com/gravitational/teleport/blob/master/integrations/access/datadog/cmd/teleport-datadog/main.go) for the standalone executable).

File: docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx
Comment (hugoShaka) [inline docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:105]: > If the recipient is a Datadog team

Can the recipient be something else than a team? If so it would be useful to define supported recipient types.

File: docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx
Comment (hugoShaka) [inline docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:109]:
```suggestion
teams attribute.

The plugin creates incidents and labels them with the recipient team. By default, the team members won't get notified, you must configure Datadog notification rules to page the team members. To do so, navigate to **Incidents -> Settings -> Notifications -> Rules** and create a new rule matching alerts labeled with the recipient team.
```

File: examples/resources/plugins/teleport-datadog.toml
Comment (hugoShaka) [inline examples/resources/plugins/teleport-datadog.toml:1]: If we only document the hosted version of the plugin, is this file used?

File: docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx
Comment (hugoShaka) [inline docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:122]: What's the format of sugested reviewrs? emails? do we also support other team names? Can datadog users have usernames that are not emails?

**Comment by bernardjkim** [comment]
Roman suggested we document both the hosted and self-hosted guide. Let me know if y'all think having two separate guides is just more confusing, or if we should try to merge them into one guide.

File: docs/config.json
Comment (r0mant) [inline docs/config.json:1720]: Not sure why these have "ssh" in the page names, I'm guessing it's an artifact of old guides or something, I'd remove it as it doesn't make sense.
```suggestion
      "source": "/access-controls/access-request-plugins/datadog-self-hosted/",
      "destination": "/admin-guides/access-controls/access-request-plugins/datadog-self-hosted/",
```

File: docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx
Comment (r0mant) [inline docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:45]: Some of these steps are exactly the same between hosted and self-hosted guides, can we move them out into includes?

**Review by r0mant — APPROVED** [review]
lgtm with a couple of comments

Do not forget to update these guides once you've added auto-approval.

File: docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx
Comment (r0mant) [inline docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:194]: Role to recipients is deprecated in favor of notification routing rules. Let's not use this anymore.

File: examples/resources/plugins/teleport-datadog.toml
Comment (r0mant) [inline examples/resources/plugins/teleport-datadog.toml:44]: As mentioned above, let's stick to notification routing rules instead of this, this is legacy configuration.

File: docs/img/enterprise/plugins/datadog/diagram.png
Comment (mmcallister) [inline docs/img/enterprise/plugins/datadog/diagram.png:1]: Looks like this is using the old Teleport logo

**Comment by hugoShaka** [comment]
> Roman suggested we document both the hosted and self-hosted guide. Let me know if y'all think having two separate guides might be more confusing, or if we should try to merge them into one guide.

I think we'll need to merge guides, especially if we do more dual "self-hosted/teleport-hosted" guides like this. The navifation is already a mess and this won't scale very well if we do this for every plugin:
image

Sorry for giving you extra work :/ (the PR content is very good though)

File: docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx
Comment (hugoShaka) [inline docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx:2]: I think the "Hosted" vs "Self-hosted" working can be improved. It's not clear for users what a hosted plugin is. Also self hosted is technically hosted by yourself, so they are not opposed concepts.

We would benefit for picking clear and well-defined terms, especially as one of our proposed Q4 goals is to align plugins in terms of feature (all access plugins should support both hosted and self-hosted modes).

WDYT of:
- Standalone vs embedded
- Standalone vs integration
- Self-hosted vs Teleport-Hosted

cc @r0mant

File: docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx
Comment (hugoShaka) [inline docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx:33]: Editor-like rights in Teleport are also required (ability to create roles and grant them to users).

File: docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx
Comment (hugoShaka) [inline docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx:142]: A user not familiar with Teleport will ask themselves "what is an AMR?", so this would benefit from a link. We don't have a nice "What is an AMR" page yet, so the most apropriate would be the AMR guide you already linked in the "Next steps".

File: examples/resources/plugins/teleport-datadog.toml
Comment (hugoShaka) [reply examples/resources/plugins/teleport-datadog.toml:44]: You still need to put the section with the catchall recipient for the plugin to start. We can edit the comment to say that AMRs are a better way of setting additional recipients.

**Comment by bernardjkim** [comment]
Okay, to limit the scope of this PR and get it merged by next week, I'll only include the "teleport-hosted" guide for this plugin. I agree that we should also include a "self-hosted" guide, but I think we want to prioritize "teleport-hosted".

Some things we might want to address before including a second "self-hosted" guide:
- The parent [Just-in-Time Access Request Plugins](https://goteleport.com/docs/admin-guides/access-controls/access-request-plugins/) page documents the plugins as "Teleport Cloud hosted" and "Self-hosted". We should decide on wording and make it consistent everywhere in the docs.
- In the parent page, these docs were originally intended only for "self-hosted" plugin guides. But it seems like it has now become a mix of "self-hosted" and "teleport-hosted". Suggestions to address this inconsistency:
  - We should organize the guides to have separate sections for "self-hosted" vs "teleport-hosted".
  - Alternatively, we remove all "teleport-hosted" guides from the documentation. Instead we improve the WebUI to include all the needed documentation and guide the user to a successful setup.
Screenshot 2024-10-11 at 10 02 16 AM

File: docs/img/enterprise/plugins/datadog/diagram.png
Comment (bernardjkim) [reply docs/img/enterprise/plugins/datadog/diagram.png:1]: Not sure how to update the Teleport logo on these diagrams. I'll just omit the diagram for now and defer this change to a later PR. We'll probably want to update all the outdated diagrams in a batch.

**Review by hugoShaka — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: docs/img/enterprise/plugins/datadog/diagram.png
Comment (mmcallister) [reply docs/img/enterprise/plugins/datadog/diagram.png:1]: Thanks Bernard! I have a PR open to work on updating all of the images using the old logo. Just didn't want to throw more in there 😁

**Pull request merged**
